From a9c0b008607ad1aaafa51cc1142dc4e67343b454 Mon Sep 17 00:00:00 2001
From: Patrick Williams <patrick@stwcx.xyz>
Date: Sat, 28 Mar 2026 08:23:09 -0400
Subject: [PATCH 17/20] oem-meta: add getSlotNumberStringByTID mapping for
 SantaBarbara

Add the slot number mapping for SantaBarbara from an early revision of
Ie5d6e6fe26baf1cb7c23d19127d0d84f66f37c27.  This moves the slot mapping
into the code dependent on MCTP configuration, which is not yet
upstream.

Signed-off-by: Patrick Williams <patrick@stwcx.xyz>
Change-Id: I7be2bb871bf6e2fec07364be1ca00e7c54586064
Upstream-Status: Pending
---
 oem/meta/utils.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/oem/meta/utils.cpp b/oem/meta/utils.cpp
index 6563a45..97f9aea 100644
--- a/oem/meta/utils.cpp
+++ b/oem/meta/utils.cpp
@@ -90,6 +90,16 @@ std::string getSlotNumberStringByTID(const pldm_tid_t& tid)
                     "TID", tid);
                 return std::to_string(slotNum);
             }
+            if (mctpEndpoint.objectPath.find("Santabarbara") !=
+                std::string::npos)
+            {
+                uint8_t slotNum = (tid / 10) - 1;
+                warning(
+                    "Fallback: Found Santabarbara in configDbusPath {CONFIG_PATH}, using slot {SLOT} for TID {TID}",
+                    "CONFIG_PATH", mctpEndpoint.objectPath, "SLOT", slotNum,
+                    "TID", tid);
+                return std::to_string(slotNum);
+            }
         }
         return std::string{"0"};
     };
-- 
2.52.0

